x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -640,14 +664,28 @@ func TestCalcSingleAssetJoin(t *testing.T) {
 
 			tokenIn := tc.tokensIn[0]
 
+			poolAssetInDenom := tokenIn.Denom
+			// when testing a case with tokenIn that does not exist in pool, we just want
+			// to provide any pool asset.
+			if tc.expErr != nil && strings.Contains(tc.expErr.Error(), doesNotExistDenom) {
```

Comment:
We should ideally use `errors.Is/As` instead of string comparisons as that is too brittle